front/src/Messages/MakeUnsafe.ts

```diff
@@ -0,0 +1,19 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import type {UnsafeString} from "./UnsafeString";
+
+type ReplaceReturnType<T extends (...a: any) => string> = (...a: Parameters<T>) => UnsafeString;
+
+/**
+ * Builds a new type where every function that returns a "string" will return a "UnsafeString" instead.
+ * This is useful to force developers into checking that a string was indeed sanitized before using it.
+ */
+export type MakeUnsafe<T> = {
+    [P in keyof T]: T[P] extends (...a: any) => string
+        ? ReplaceReturnType<T[P]>
+        : T[P] extends (...a: any) => any
+            ? (...a: Parameters<T[P]>) => ReturnType<T[P]>
```

Comment:
use MakeUnsafe on ReturnType

Reply:
here is my attempt 
additional advantage you can also use it directly on string types
```
type ReplaceReturnType<T extends (...args) => any> = (...a: Parameters<T>) => MakeUnsafe<ReturnType<T>>;


export type MakeUnsafe<T> = T extends String
    ? UnsafeString
    : (T extends Array<infer SubType>
        ? Array<MakeUnsafe<SubType>>
        : (T extends (...args) => any
            ? ReplaceReturnType<T>
            : {
                [P in keyof T]: MakeUnsafe<T[P]>
            }))

```

Reply:
@moufmouf

Reply:
:+1:

Reply:
Your attempt seems to loop for some reason:

```
front_1          | ERROR in src/Connexion/RoomConnection.ts:528:35
front_1          | TS2589: Type instantiation is excessively deep and possibly infinite.
front_1          |     526 |                 event: message.getEvent(),
front_1          |     527 |                 parameters: JSON.parse(message.getParametersjson()),
front_1          |   > 528 |                 state: JSON.parse(message.getStatejson())
front_1          |         |                                   ^^^^^^^^^^^^^^^^^^^^
front_1          |     529 |             });
front_1          |     530 |         });
front_1          |     531 |     }
```

Not sure why yet.

Reply:
🤔 probably some subtype i missed - maybe Date ?

Reply:
oh i found it 
 ? ReplaceReturnType<T> instead of MakeUnsafe<T>

Reply:
```
type ReplaceReturnType<T extends (...args: unknown[]) => unknown> = (...a: Parameters<T>) => MakeUnsafe<ReturnType<T>>;

type Primitives = undefined | null | boolean | number

export type MakeUnsafe<T> = T extends Primitives ? T : T extends String
    ? UnsafeString
    : (T extends Array<infer SubType>
        ? Array<MakeUnsafe<SubType>>
        : (T extends (...args: unknown[]) => unknown
            ? ReplaceReturnType<T>
            : {
                [P in keyof T]: MakeUnsafe<T[P]>
            }))

```

alse removed the any

Reply:
Beautiful!